projects/admin-core/src/lib/application/state/application.reducer.ts

```diff
@@ -360,6 +360,79 @@ const onUpdateApplicationFiltersConfig = (
   }));
 };
 
+const onCreateApplicationFilterGroup = (
+  state: ApplicationState,
+  payload: ReturnType<typeof ApplicationActions.createApplicationAttributeFilter>,
+): ApplicationState => {
+  return updateApplication(state, application => {
+    let addedToExistingGroup = false;
+    const newFilterGroups = application.settings?.filterGroups?.map(filterGroup => {
+      if (payload.filterGroup.layerIds.every(layerId => filterGroup.layerIds.includes(layerId))
+        && payload.filterGroup.layerIds.length === filterGroup.layerIds.length) {
```

Comment:
For efficiency its better to compare length first, this is an easy comparison and if inequal no need check every